### DISCUSSION TITLE
Serving only needed files

### DISCUSSION THREAD

**Discussion by alouini333** [nuxt/nuxt#9647] [Questions (Nuxt 2)] [answered]
Recently I developed a Nuxtjs application.
A security audit was taken place in the company and we found out that the logic of the application is visible to non authenticated user.

That means when the application is loaded in the browser and someone opens the Dev tools, he can access the app.js file and see the logic, the different api routes without being autehnticated.

This was considered as a critical issue, and we are looking to fix this issue.

I started investigating,

First thing is adding an obfuscator to the webpack config to make the code unreadable, but somehow the code in the dist/_nuxt/app.js is still visible and readable (minified but readable).

So currently, I'm looking for a solution, when it makes it possible to turn the generated app.js file into chunks and serve them to the browser only when needed.

Best regards
Amine

**Comment by TheAlexLichter** [comment, marked as answer]
This is something you can't avoid eventually. The front end will always have it's full logic exposed in one way or another. Even if you split the `app.js` in smaller files (which you should - check out "code splitting"), webpack will have a so-called manifest (think of a "lookup file") that includes the name of all JavaScript chunks/files.

The "real" logic for authenticated users should be part of your backend and should only be accessible to authenticated users (e.g. by a authentication token).

**Reply by ghost** [reply]
Was having the same problem. Thanks a lot!
